server/src/main/java/org/elasticsearch/action/admin/cluster/stats/MappingStats.java

```diff
@@ -170,16 +199,51 @@ private static int countOccurrences(String script, Pattern pattern) {
     }
 
     MappingStats(StreamInput in) throws IOException {
+        if (in.getVersion().onOrAfter(Version.V_8_4_0)) {
+            totalFieldCount = in.readOptionalVLong();
+            totalDeduplicatedFieldCount = in.readOptionalVLong();
+            totalMappingSizeBytes = in.readOptionalVLong();
+        } else {
+            totalFieldCount = null;
+            totalDeduplicatedFieldCount = null;
+            totalMappingSizeBytes = null;
+        }
         fieldTypeStats = Collections.unmodifiableList(in.readList(FieldStats::new));
         runtimeFieldStats = Collections.unmodifiableList(in.readList(RuntimeFieldStats::new));
     }
 
     @Override
     public void writeTo(StreamOutput out) throws IOException {
+        if (out.getVersion().onOrAfter(Version.V_8_4_0)) {
+            out.writeOptionalVLong(totalFieldCount);
+            out.writeOptionalVLong(totalDeduplicatedFieldCount);
+            out.writeOptionalVLong(totalMappingSizeBytes);
+        } // else just omit these stats, they're not computed on older nodes anyway
         out.writeCollection(fieldTypeStats);
         out.writeCollection(runtimeFieldStats);
     }
 
+    /**
+     * @return the total number of fields (in non-system indices).
+     */
+    public long getTotalFieldCount() {
+        return totalFieldCount;
+    }
+
+    /**
+     * @return the total number of fields (in non-system indices) accounting for deduplication.
+     */
+    public long getTotalDeduplicatedFieldCount() {
+        return totalDeduplicatedFieldCount;
+    }
+
+    /**
+     * @return the total size of all mappings (including those for system indices) accounting for deduplication and compression.
+     */
+    public long getTotalMappingSizeBytes() {
+        return totalMappingSizeBytes;
+    }
```

Comment:
Should above getters return `Long` to avoid NPE?

Reply:
++ or `OptionalLong`, even better